esphome/components/lvgl/widgets/spinbox.py

```diff
@@ -38,18 +39,24 @@ def validate_spinbox(config):
     min_val = -1 - max_val
     range_from = int(config[CONF_RANGE_FROM])
     range_to = int(config[CONF_RANGE_TO])
-    step = int(config[CONF_STEP])
+    step = config.get(CONF_SELECTED_DIGIT)
+    digits = config[CONF_DIGITS]
     if (
         range_from > max_val
         or range_from < min_val
         or range_to > max_val
         or range_to < min_val
     ):
-        raise cv.Invalid("Range outside allowed limits")
-    if step <= 0 or step >= (range_to - range_from) / 2:
-        raise cv.Invalid("Invalid step value")
-    if config[CONF_DIGITS] <= config[CONF_DECIMAL_PLACES]:
-        raise cv.Invalid("Number of digits must exceed number of decimal places")
+        raise cv.Invalid("Range outside allowed limits", path=[CONF_RANGE_FROM])
+    if digits <= config[CONF_DECIMAL_PLACES]:
+        raise cv.Invalid(
+            "Number of digits must exceed number of decimal places", path=[CONF_DIGITS]
+        )
+    if step is not None and step >= digits:
```

Comment:
Here too

Reply:
the default 0 would be falsey if you need to check it
```suggestion
    if step and step >= digits:
```